File: backend/src/apps/api/serializers/clients.py
from rest_framework import serializers
from apps.clients.models import Client
from apps.core.serializers import BaseOrganizationModelSerializer


class ClientSerializer(BaseOrganizationModelSerializer):
    phone = serializers.RegexField(regex=r"^(\+7|8)\d{10}$")

    class Meta:
        model = Client
        fields = (
            "id",
            "first_name",
            "last_name",
            "middle_name",
            "phone",
            "email",
            "created_at",
            "updated_at",
        )

File: frontend/src/api/clients.ts
import api from "~/lib/axios";

export interface Client {
  id: string;
  first_name: string;
  last_name: string;
  middle_name?: string;
  phone: string;
  email?: string;
  created_at: string;
  updated_at: string;
}

export async function getClients(orgId: number) {
  return api.get<Client[]>(`/v1/organizations/${orgId}/clients/`).then((res) => res.data);
}
